beacon-chain/state/v1/setters_block.go

```diff
@@ -52,14 +52,11 @@ func (b *BeaconState) UpdateBlockRootAtIndex(idx uint64, blockRoot [32]byte) err
 
 	r := b.state.BlockRoots
 	if ref := b.sharedFieldReferences[blockRoots]; ref.Refs() > 1 {
-		// Copy elements in underlying array by reference.
-		r = make([][]byte, len(b.state.BlockRoots))
-		copy(r, b.state.BlockRoots)
```

Comment:
Please review that this is OK

Reply:
no it isn't, this will lead to a corrupted state removing this copy.